samples/bluetooth/unicast_audio_client/src/main.c

```diff
@@ -38,39 +38,154 @@ static K_SEM_DEFINE(sem_stream_qos, 0, 1);
+ * @param frequency_hz frequency in Hz
+ * @param sample_rate_hz sample-rate in Hz.
+ */
+static void fill_audio_buf_sin(int16_t *buf, int length_us, int frequency_hz, int sample_rate_hz)
 {
-	while (len-- != 0) {
-		printk("%02x", *ptr++);
+	int sine_period = sample_rate_hz/frequency_hz;
+	int num_samples = (length_us * sample_rate_hz) / 1000000;
+	float step = 3.1415 / sine_period;
+	float sample;
+
+	for (int i = 0; i < num_samples; i++) {
+		sample = sin(i * step);
+		buf[i] = (int16_t)(INT16_MAX * sample);
+		/* printk("%3i: rad: %i sample: %i %6i\n", i, (int)(rad*1000), (int)(sample*1000),
+		 * (int16_t)(sample * INT16_MAX));
+		 */
 	}
 }
 
-static void print_codec(const struct bt_codec *codec)
+static const char *getErrString(uint16_t err)
 {
-	printk("codec 0x%02x cid 0x%04x vid 0x%04x count %u\n",
-	       codec->id, codec->cid, codec->vid, codec->data_count);
+	switch (err) {
+	case LC3_ENCODE_ERR_FREE:
+		return "LC3_ENCODE_ERR_FREE";
+	case LC3_ENCODE_ERR_INVALID_CONFIGURATION:
+		return "LC3_ENCODE_ERR_INVALID_CONFIGURATION";
+	case LC3_ENCODE_ERR_INVALID_BYTE_COUNT:
+		return "LC3_ENCODE_ERR_INVALID_BYTE_COUNT";
+	case LC3_ENCODE_ERR_INVALID_BITS_PER_AUDIO_SAMPLE:
+		return "LC3_ENCODE_ERR_INVALID_BITS_PER_AUDIO_SAMPLE";
+	case LC3_ENCODE_ERR_ENCODER_ALLOCATION:
+		return "LC3_ENCODE_ERR_ENCODER_ALLOCATION";
+	}
+	return "UNKNOWN LC3 ENCODE ERROR";
+}
 
-	for (size_t i = 0; i < codec->data_count; i++) {
-		printk("data #%zu: type 0x%02x len %u\n",
-		       i, codec->data[i].data.type,
-		       codec->data[i].data.data_len);
-		print_hex(codec->data[i].data.data,
-			  codec->data[i].data.data_len -
-				sizeof(codec->data[i].data.type));
-		printk("\n");
+static void lc3_audio_timer_timeout(struct k_work *work)
+{
+	/* For the first call-back we push multiple audio frames to the buffer to use the
+	 * controller ISO buffer to handle jitter.
+	 */
+	static int prime_count = 2;
+	static int64_t start_time;
+	static int32_t pdu_cnt;
+	int32_t pdu_goal_cnt;
+	int64_t uptime, run_time_ms;
+	int ret;
+	struct net_buf *buf;
+	uint8_t *net_buffer;
+	uint16_t lc3_ret;
+	uint16_t tx_sdu_len = audio_stream.iso->qos->tx->sdu;
+
+	k_work_schedule(&audio_send_work, K_MSEC(10));
+
+	if (start_time == 0) {
+		/* Read start time and produce the number of frames needed to catch up with any
+		 * inaccuracities in the timer. by calculating the number of frames we should
+		 * have send and compare to how many wa have actually sent.
+		 */
+		start_time = k_uptime_get();
 	}
 
-	for (size_t i = 0; i < codec->meta_count; i++) {
-		printk("meta #%zu: type 0x%02x len %u\n",
-		       i, codec->meta[i].data.type,
-		       codec->meta[i].data.data_len);
-		print_hex(codec->meta[i].data.data,
-			  codec->meta[i].data.data_len -
-				sizeof(codec->meta[i].data.type));
-		printk("\n");
+	uptime = k_uptime_get();
+	run_time_ms = uptime - start_time;
+	pdu_goal_cnt = ((run_time_ms * 10) / (AUDIO_LENGTH_US/100)); /* scale to allow 7.5ms */
```

Comment:
```suggestion
	pdu_goal_cnt = ((run_time_ms * 10) / (AUDIO_LENGTH_US / 100)); /* scale to allow 7.5ms */
```

Reply:
I'm not really sure how this scales to 7.5ms; looks more like it is converting it into 10ms units